distributed/scheduler.py

```diff
@@ -823,8 +823,11 @@ def _dec_needs_replica(self, ts: TaskState) -> None:
         if self.needs_what[ts] == 0:
             del self.needs_what[ts]
             nbytes = ts.get_nbytes()
-            self._network_occ -= nbytes
-            self.scheduler._network_occ_global -= nbytes
+            # FIXME: ts.get_nbytes may change if non-deterministic tasks get recomputed, causing drift
+            self._network_occ -= min(nbytes, self._network_occ)
```

Comment:
is network_occ an integer or are we dealing with floating point foo at this point?

Reply:
looks like `_network_occ_global` is defined as a float. I guess this won't be a drama since we only add and subtract integers.

Reply:
I've adjusted types and initial values to match the assumption of integer values.